### ISSUE TITLE
Implement Bierbot Tricks software for Sonoff ESP32

### ISSUE THREAD

**Issue by humpejunior** [thorrak/brewflasher#30]
I'd like to flash the software from BernhardSchlegel (Brick-23) from git on an ESP-32 and thought, that this would be the right software for that.
Maybe add that as an option?

Best wishes from Germany.

P.S.: I could also provide a better translation into German... :)

**Comment by thorrak** [comment]
Can you provide a link to the project? I tend to try not to add projects without the consent/support of the author, but I’m happy to reach out to him/her to see if I could add it!

With regards to the translation, if you happen to be on HomeBrewTalk send me a PM there and I can get you set up. ;)

I wish there was a better way to do user-to-user direct communication on GitHub, but alas…

**Comment by thorrak** [comment]
I think I found an email address for the author. I’ve reached out - we’ll see if he responds.

**Comment by BernhardSchlegel** [comment]
Author responded and is honored ! Looking forward to the collab :)

**Comment by humpejunior** [comment]
Sorry for not responding immediately...
I'm quite new to GitHub and that's my first "issue". I'm still figuring out how all this works...:)

Great to hear, that Bernhard likes to work together!
Finally a way to flash an esp with a mac

**Comment by thorrak** [comment]
Yes he did - that email address apparently worked! :)

@BernhardSchlegel - I'll send you a note here shortly with everything you need to know. 

@humpejunior - No worries at all - and thanks for the introduction! Bierbot Bricks sounds like a really interesting project, and wasn't one I was familiar with before you mentioned it!

**Comment by humpejunior** [comment]
No problem. I always try to mention that project. :D It's such an awesome way to automate brewing, stirring, fermenting and so on...

**Comment by BernhardSchlegel** [comment]
Just FYI upfront, there are actually two implementations which do the same but support different chips:

- ESP8266: https://github.com/BernhardSchlegel/BierBot-Bricks
- ESP32: https://github.com/BernhardSchlegel/Brick-32

**Comment by humpejunior** [comment]
@BernhardSchlegel Yeah good, that you mention this. I have an old TH Origin laying around and recently purchased a TH320 and wanted to flash it with a MacBook but wasn't able to find an ESP flash download tool for Mac that supports ESP32 chips...

**Comment by thorrak** [comment]
All good! It's your call in BrewFlasher if you want to treat them as two separate projects, or combine them together and let the user select which controller type to flash.

I went ahead and got projects created in BrewFlasher for both, and was able to successfully link the BrewBot Bricks (ESP8266) [firmware](https://web.brewflasher.com/fw/150). If you have a moment, @BernhardSchlegel , there's one thing I need your help in doing in order to set up the ESP32 firmware. I sent you an email with more information - let me know if it makes sense or if you have any questions.

**Comment by Nakita2k** [comment]
How do you set the Tilt Color with Brewflasher Repeater ?

**Comment by thorrak** [comment]
@Nakita2k  Since your question isn’t related to BierBot Bricks I went ahead and moved it here: https://github.com/thorrak/brewflasher/issues/37

**Comment by BernhardSchlegel** [comment]
Hey there, I think the [new Release](https://github.com/BernhardSchlegel/Brick-32/releases/tag/v.0.9.0) should solve the issue. Now even supporting both TH Elite (with a display) and TH Origin (if you don't need one).

![Image](https://github.com/user-attachments/assets/86e868a6-29fb-4045-a9d6-8a0fbe50a633)

Let me know if you have trouble, maybe even over [here](https://github.com/BernhardSchlegel/Brick-32/issues).
